spec/forms/candidate_interface/reference/request_referee_email_address_form_spec.rb

```diff
@@ -0,0 +1,95 @@
+require 'rails_helper'
+
+RSpec.describe CandidateInterface::Reference::RequestRefereeEmailAddressForm, type: :model do
+  describe 'validations' do
+    let(:form) { subject }
+
+    context 'when a duplicate email is given' do
+      let!(:application_form) { create(:application_form) }
+      let!(:application_reference) { create(:reference, email_address: nil, application_form:) }
+
+      context 'when duplicate reference is not sent' do
+        it 'returns custom error message' do
+          create(:reference, :not_requested_yet, email_address: 'iamtheone@whoknocks.com', application_form:)
+          form = described_class.new(email_address: 'iAMtheone@whoknocks.com', reference_id: application_reference.id)
+
+          form.save(application_reference)
+          expect(form.errors[:email_address]).to include('A reference request has already been started for this email address')
+        end
+      end
```

Comment:
could extract these five specs to a shared example?

Reply:
Done!